frontend/pages/admin/IntegrationsPage/cards/CertificateAuthorities/components/EditCertAuthorityModal/helpers.tsx

```diff
@@ -38,9 +38,11 @@ export const generateDefaultFormData = (
       apiToken: certAuthority.api_token,
       profileId: certAuthority.profile_id,
       commonName: certAuthority.certificate_common_name,
-      userPrincipalName: certAuthority.certificate_user_principal_names
-        ? certAuthority.certificate_user_principal_names[0]
-        : "",
+      userPrincipalName:
+        certAuthority.certificate_user_principal_names !== null &&
+        certAuthority.certificate_user_principal_names.length > 0
+          ? certAuthority.certificate_user_principal_names[0]
+          : "",
```

Comment:
Could this be improved to `userPrincipalName: certAuthority.certificate_user_principal_names?.[0] || "",`